### PR TITLE
Change password flow

### PR DESCRIPTION
**Pull request by TuxBoy**
A small contribution for the issues #12

### CODE REVIEW

**Comment by brendt** [comment]
I was like: "11.000 lines changes?!?" This was before I saw an update to `package.lock` 😄 I'm going to take a look at this soon!

File: app/App/Http/Controllers/UserProfileController.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        return (new ProfileViewModel($user))->view('userProfile.index');
+    }
+
+    public function update(
+        User $user,
+        ProfileRequest $request,
+        ChangePasswordAction $changePasswordAction
+    ): RedirectResponse {
+        $changePasswordAction($user, $request->input('password'));
```
Comment (brendt) [inline app/App/Http/Controllers/UserProfileController.php]: I think we should pass the hashed password to the action. I want to work with the hash as soon as possible, avoiding security issues in the future.

Say we'd make this action queuable (that's not likely to happen, I know); the plain password would be stored on the sever somewhere. We don't want that 😄

There's two ways we could approach this: hash the password in the controller, or create a DTO solely for passwords: `PasswordData`, which has a `$password_hash` property, and immediately hashes the password via a static constructor. For example: `PasswordData::fromPlainString($request->input('password'))`

What do you think?

File: app/App/Http/ViewModels/ProfileViewModel.php [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+namespace App\Http\ViewModels;
+
+use Spatie\ViewModels\ViewModel;
+
+class ProfileViewModel extends ViewModel
```
Comment (brendt) [inline app/App/Http/ViewModels/ProfileViewModel.php]: I believe you forget the constructor here. According to the controller, it should take a User.

Simply setting the user as a public property on the view model will be enough for it to end up as a variable in the view, or you could make `ProfileViewModel::$user` private, and expose is via a public `ProfileViewModel::user()` method.

Comment (TuxBoy) [reply app/App/Http/ViewModels/ProfileViewModel.php]: Do you prefer to use a getter or expose the properties in public?

Comment (brendt) [reply app/App/Http/ViewModels/ProfileViewModel.php]: I prefer the getter approach 😄

Comment (TuxBoy) [reply app/App/Http/ViewModels/ProfileViewModel.php]: Me too ;)

**Pull request merged**